### PR TITLE
Add option to use testing apt repos for ros

### PR DESCRIPTION
**Pull request by emersonknapp**
Signed-off-by: Emerson Knapp <emerson.b.knapp@gmail.com>

### CODE REVIEW

File: ros_cross_compile/platform.py [hunk 1]
```diff
@@ -61,7 +61,10 @@ class Platform:
     """
 
     def __init__(
-        self, arch: str, os_name: str, ros_distro: str, override_base_image: Optional[str] = None,
+        self,
+        arch: str, os_name: str, ros_distro: str,
+        override_base_image: Optional[str] = None,
+        testing_version: bool = False,
```
Comment (zmichaels11) [inline ros_cross_compile/platform.py:67]: Is there a reason why some are on separate lines and not all/none?

Comment (zmichaels11) [reply ros_cross_compile/platform.py:67]: Even though it is a boolean statement, I think `testing_version` would be more human-readable as an enum.
ex:
`Platform('x86_64', 'debian', 'melodic', VersionType.TESTING)`

File: ros_cross_compile/platform.py [hunk 2]
```diff
@@ -61,7 +61,13 @@ class Platform:
...
 
     def __init__(
-        self, arch: str, os_name: str, ros_distro: str, override_base_image: Optional[str] = None,
+        self,
+        arch: str,
+        os_name: str,
+        ros_distro: str,
+        *,
+        override_base_image: Optional[str] = None,
+        testing_version: bool = False,
```
Comment (thomas-moulard) [inline ros_cross_compile/platform.py:70]: testing_version is not really clear. It seems we're enabling testing of the version of... something?

`use_shadow_testing_repository` for instance?

File: ros_cross_compile/platform.py [hunk 3]
```diff
@@ -79,6 +85,8 @@ def __init__(
             self._ros_version = 'ros'
         else:
             raise ValueError('Unknown ROS distribution "{}" specified'.format(ros_distro))
+        if testing_version:
+            self._ros_version = '{}-testing'.format(self._ros_version)
```
Comment (thomas-moulard) [inline ros_cross_compile/platform.py:89]: I don't think that passing a boolean is necessarily the best, it seems needlessly constraining users. Why can't users just pass one or more extra APT repos that would be added to the Docker sysroot image? That would prevent cross-compile from breaking if the repo structure changes.

**Comment by emersonknapp** [comment]
@thomas-moulard  here's where the dockerfile mentions the repo https://github.com/ros-tooling/cross_compile/pull/197/files?file-filters%5B%5D=.Dockerfile#diff-2fa487816a196f4a8a73123c48758f2fR40 - i was just relying on $ROS_VERSION to get it done.

I see your point about enabling arbitrary extra repositories. Now that you mention that, this same result is already possible using the custom-setup-script - the user can execute arbitrary code before the rosdeps are installed, so they could just add extra apt lists to check. Instead of adding this feature, I can call out that use case in the README

**Pull request closed without merging**